Sources/Identity/CustomerInfo+NonSubscriptions.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  CustomerInfo+NonSubscriptions.swift
+//
+//  Created by Nacho Soto on 7/18/23.
+
+import Foundation
+
+extension CustomerInfo {
+
+    func containsNonSubscription(_ transation: StoreTransactionType) -> Bool {
+        return self.nonSubscriptions.contains {
+            $0.transactionIdentifier == transation.transactionIdentifier
```

Comment:
@tonycosentini this is ready. Just gotta change to use the real store transaction ID not the RC ID.
Also I wonder if we should restrict this by sandbox and store too /cc @aboedo

Reply:
Does non subscription here mean both a consumable and a non-consumable? 🤔 

Can the answer to this be added as a comment to this function?

Reply:
TODO: we discovered that they're included, so test those too.

Reply:
I updated the PR, logs, and code to make this more clear. I was mixing consumables with non-subscriptions everywhere.

> We might wanna do it for non-renewing subscriptions as well, since they have the same characteristics as consumables in that they disappear from the receipt forever.
> 
> And lastly, non-consumables don't actually need this, do they get the same treatment?

See `TransactionPoster.shouldFinish(transaction:for:customerInfo:)`. That new logic is based on `StoreProduct.productCategory` (subscription / non-subscriptions). We can't use `ProductType` because we can't determine that for `SK1` products (see `SK1StoreProduct.productType`).

Reply:
Just a note that we _could_ in theory distinguish between non-renewing and auto-renewing subs by mixing in the subs vs non subs distinction + expiration date != nil, but it's a non-trivial change that feels a bit more risky, and it isn't worth the effort.